scripts/synchronize-knative-manifests.sh

```
@@ -1,62 +1,40 @@
+
+setup_error_handling
+
+COMPONENT_NAME="knative"
 KN_SERVING_RELEASE="v1.16.2" # Must be a release
 KN_EXTENSION_RELEASE="v1.16.0" # Must be a release
 KN_EVENTING_RELEASE="v1.16.4" # Must be a release
-BRANCH=${BRANCH:=synchronize-knative-manifests-${KN_SERVING_RELEASE?}}
+BRANCH=${BRANCH:=synchronize-${COMPONENT_NAME}-manifests-${KN_SERVING_RELEASE?}}
 
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+# Path configurations
 MANIFESTS_DIR=$(dirname $SCRIPT_DIR)
+DST_DIR=$MANIFESTS_DIR/common/${COMPONENT_NAME}
 
-# replace source regex ($1) with target regex ($2)
-# in file ($3)
-replace_in_file() {
-  SRC_TXT=$1
-  DST_TXT=$2
-  sed -i "s|$SRC_TXT|$DST_TXT|g" $3
-}
-
-echo "Creating branch: ${BRANCH}"
+create_branch "$BRANCH"
 
-if [ -n "$(git status --porcelain)" ]; then
-  echo "WARNING: You have uncommitted changes"
-fi
-if [ `git branch --list $BRANCH` ]
-then
-   echo "WARNING: Branch $BRANCH already exists."
-fi
+check_uncommitted_changes
 
-# Create the branch in the manifests repository
-if ! git show-ref --verify --quiet refs/heads/$BRANCH; then
-    git checkout -b $BRANCH
-else
-    echo "Branch $BRANCH already exists."
-fi
-
-if [ -n "$(git status --porcelain)" ]; then
-  echo "WARNING: You have uncommitted changes"
-fi
-
-DST_DIR=$MANIFESTS_DIR/common/knative
+# Clean up existing files (keep README and OWNERS)
 if [ -d "$DST_DIR" ]; then
-    # keep README and OWNERS file
-    rm -r "$DST_DIR/knative-serving/base/upstream"
-    rm "$DST_DIR/knative-serving-post-install-jobs/base/serving-post-install-jobs.yaml"
-    rm -r "$DST_DIR/knative-eventing/base/upstream"
-    rm "$DST_DIR/knative-eventing-post-install-jobs/base/eventing-post-install.yaml"
+    rm -r "$DST_DIR/knative-serving/base/upstream" 2>/dev/null || true
+    rm "$DST_DIR/knative-serving-post-install-jobs/base/serving-post-install-jobs.yaml" 2>/dev/null || true
+    rm -r "$DST_DIR/knative-eventing/base/upstream" 2>/dev/null || true
```

Why do you add 2>/dev/null || true ? Let it always fail hard.